examples/tutorials/core_api/1_metrics.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python3
+
+"""
+Stage 1: Let's create the core_context, and use it to start logging metrics.
+This is just a few lines of code, and we'll be able to see the results in the
+Determined WebUI.
+"""
+
+import logging
+import sys
+import time
+
+# NEW: import determined
+import determined as det
+
+def main(core_context, increment_by):
+    x = 0
+    for batch in range(100):
+        x += increment_by
+        time.sleep(.1)
+        print("x is now", x)
+        # NEW: report training metrics to Determined.
+        if batch % 10 == 9:
+            core_context.train.report_training_metrics(latest_batch=batch, metrics={"x": x})
```

Comment:
it'd seem more aligned if x was reported at the same frequency as `print`?

Reply:
Do you mean `print` is too frequent? I would agree with that.

Reply:
I mean it'd be nice to have 1:1 mapping between manual prints and the metrics "showing up in the webui" - should make it clearer where these are coming from.

Reply:
it's a bit tricky for me to guess Ryan's intent. But I think the idea was to make explicit the "learning" - processing of individual batches. You will notice that 0_start does not collect metrics at all:
```
[2022-04-25T18:48:44.442200Z] bbedc350 || INFO: [1] root: Launching: ['sh', '-c', './0_start.py']
[2022-04-25T18:48:44.567337Z] bbedc350 || x is now 1
[2022-04-25T18:48:44.672570Z] bbedc350 || x is now 2
[2022-04-25T18:48:44.774656Z] bbedc350 || x is now 3
[2022-04-25T18:48:44.879599Z] bbedc350 || x is now 4
[2022-04-25T18:48:44.981441Z] bbedc350 || x is now 5
[2022-04-25T18:48:45.082459Z] bbedc350 || x is now 6
[2022-04-25T18:48:45.184186Z] bbedc350 || x is now 7
[2022-04-25T18:48:45.285437Z] bbedc350 || x is now 8
[2022-04-25T18:48:45.388754Z] bbedc350 || x is now 9
[2022-04-25T18:48:45.491606Z] bbedc350 || x is now 10
[2022-04-25T18:48:45.595408Z] bbedc350 || x is now 11
```
The next step, 1_metrics, keeps printing the value of x at every "batch" but in addition reports the metrics every 10 batches:
```
[2022-04-25T18:49:19.250446Z] 87c021c1 || INFO: [1] root: Launching: ['sh', '-c', './1_metrics.py']
[2022-04-25T18:49:20.788894Z] 87c021c1 || x is now 1
[2022-04-25T18:49:20.891923Z] 87c021c1 || x is now 2
[2022-04-25T18:49:20.995577Z] 87c021c1 || x is now 3
[2022-04-25T18:49:21.099430Z] 87c021c1 || x is now 4
[2022-04-25T18:49:21.204740Z] 87c021c1 || x is now 5
[2022-04-25T18:49:21.308930Z] 87c021c1 || x is now 6
[2022-04-25T18:49:21.411429Z] 87c021c1 || x is now 7
[2022-04-25T18:49:21.516599Z] 87c021c1 || x is now 8
[2022-04-25T18:49:21.619109Z] 87c021c1 || x is now 9
[2022-04-25T18:49:21.720999Z] 87c021c1 || x is now 10
[2022-04-25T18:49:21.721534Z] 87c021c1 || INFO:determined.core:report_training_metrics(latest_batch=9, metrics={'x': 10})
[2022-04-25T18:49:21.868868Z] 87c021c1 || x is now 11
[2022-04-25T18:49:21.971171Z] 87c021c1 || x is now 12
```

Reply:
If you are not convinced, I can reduce the frequency to 1 in 10.

Reply:
okay, maybe I am overthinking this.